### PR TITLE
Fix Adaptive Banner and update example app

### PR DESCRIPTION
**Pull request by bparrishMines**
## Description

Fixed decoding Adaptive banner ad size and update example app with the new size.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/googleads/googleads-mobile-flutter/issues
[Contributor Guide]: https://github.com/googleads/googleads-mobile-flutter/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/google_mobile_ads/example/lib/reusable_inline_example.dart [hunk 1]
```diff
@@ -23,52 +37,47 @@ class _ReusableInlineExampleState extends State<ReusableInlineExample> {
...
-            padding: const EdgeInsets.symmetric(horizontal: 16.0),
-            child: ListView.separated(
-              cacheExtent: 500,
-              itemCount: 20,
-              separatorBuilder: (BuildContext context, int index) {
+  Widget build(BuildContext context) => Center(
+        child: Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 16.0),
+          child: ListView.separated(
+            itemCount: 30,
```
Comment (jjliu15) [inline packages/google_mobile_ads/example/lib/reusable_inline_example.dart:44]: Is it okay for itemCount to be 30? This allows for the same adwidget to be returned for multiple indexes. This could potentially cause a bug if the app was built on an ipad.

File: packages/google_mobile_ads/test/ad_containers_test.dart [hunk 2]
```diff
@@ -880,3 +880,17 @@ void main() {
     });
   });
 }
+
+class TestMessageCodec extends AdMessageCodec {
```
Comment (jjliu15) [inline packages/google_mobile_ads/test/ad_containers_test.dart:884]: Why is this necessary?

File: packages/google_mobile_ads/ios/Tests/FLTGoogleMobileAdsReaderWriterTest.m [hunk 3]
```diff
@@ -218,3 +227,35 @@ - (GADAdSize)landscapeAnchoredAdaptiveBannerAdSizeWithWidth:(NSNumber *)width {
...
+@end
+
+@implementation TestGoogleMobileAdsReader
+- (instancetype _Nonnull)initWithFactory:(FLTAdSizeFactory *_Nonnull)adSizeFactory
+                                    data:(NSData *_Nonnull)data {
+  return [super initWithFactory:adSizeFactory data:data];
+}
+
+- (id _Nullable)readValueOfType:(UInt8)type {
+  FLTAdMobField field = (FLTAdMobField)type;
```
Comment (jjliu15) [inline packages/google_mobile_ads/ios/Tests/FLTGoogleMobileAdsReaderWriterTest.m:249]: Can you add a comment explaining why this override is necessary?

File: packages/google_mobile_ads/example/lib/main.dart [hunk 4]
```diff
@@ -39,78 +44,104 @@ class _MyAppState extends State<MyApp> {
...
+        onAdFailedToLoad: (Ad ad, LoadAdError error) {
+          print('$BannerAd failedToLoad: $error');
+          ad.dispose();
+        },
+        onAdOpened: (Ad ad) => print('$BannerAd onAdOpened.'),
+        onAdClosed: (Ad ad) => print('$BannerAd onAdClosed.'),
+        onApplicationExit: (Ad ad) => print('$BannerAd onApplicationExit.'),
+      ),
+    );
+    return banner.load();
```
Comment (jjliu15) [inline packages/google_mobile_ads/example/lib/main.dart:97]: You should keep a reference to the banner ad and call `dispose` when it's no longer needed.

File: packages/google_mobile_ads/example/lib/reusable_inline_example.dart (hunk 1 above)
Comment (bparrishMines) [reply packages/google_mobile_ads/example/lib/reusable_inline_example.dart]: That's a good point. I change the Ad indexes back to 5, 10, 15.

File: packages/google_mobile_ads/test/ad_containers_test.dart (hunk 2 above)
Comment (bparrishMines) [reply packages/google_mobile_ads/test/ad_containers_test.dart]: removed

File: packages/google_mobile_ads/example/lib/main.dart (hunk 4 above)
Comment (bparrishMines) [reply packages/google_mobile_ads/example/lib/main.dart:97]: Banner is stored in `onAdLoaded` and disposed in `onAdFailedToLoad` callbacks

Comment (jjliu15) [reply packages/google_mobile_ads/example/lib/main.dart:97]: Since we create a new `BannerAd` here whenever `MyAppState` is initialized, I think it's better practice to call `_anchoredBanner?.dispose()` from `_MyAppState.dispose()`. This might not be necessary in the particular case of the example app,  since `_MyAppState.dispose()` seems to only be called when the app is killed.

Comment (bparrishMines) [reply packages/google_mobile_ads/example/lib/main.dart:97]: Ah, I meant to include it in there as well. I added it.

Comment (jjliu15) [reply packages/google_mobile_ads/example/lib/main.dart:97]: Nevermind - I forgot that AdWidget calls ad.dispose()

File: packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/AdMessageCodec.java [hunk 5]
```diff
@@ -195,7 +195,6 @@ protected void writeAdSize(ByteArrayOutputStream stream, FlutterAdSize value) {
           (FlutterAdSize.AnchoredAdaptiveBannerAdSize) value;
       writeValue(stream, size.orientation);
       writeValue(stream, size.width);
-      writeValue(stream, size.height);
```
Comment (jjliu15) [inline packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/AdMessageCodec.java:198]: Is this still necessary? Since we directly pass back the height now in `GoogleMobileAdsPlugin`

Comment (bparrishMines) [reply packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/AdMessageCodec.java:198]: Yes, but only for testing. Its the same for the other `AdSize`s in the `AdMessageCodec`. The Dart code of the library only encodes and sends `AdSize`s, but there is support to decode it so that its easier to test.

**Review by jjliu15 — APPROVED** [review]

File: packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/Constants.java [hunk 6]
```diff
@@ -17,5 +17,5 @@
 /** Constants used in the plugin. */
 public class Constants {
   /** Version request agent. Should be bumped alongside plugin versions. */
-  public static final String REQUEST_AGENT_PREFIX_VERSIONED = "Flutter-GMA-0.12.2";
+  public static final String REQUEST_AGENT_PREFIX_VERSIONED = "Flutter-GMA-0.12.2+1";
```
Comment (blasten) [inline packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/Constants.java:20]: >Version request agent. Should be bumped alongside plugin versions.

do we have a test for this?
